Source/Core/VideoBackends/D3D/D3DBoundingBox.cpp

```diff
@@ -67,23 +81,73 @@ void BBox::Shutdown()
   s_bbox_buffer.Reset();
 }
 
-void BBox::Set(int index, int value)
+void BBox::Flush()
 {
-  D3D11_BOX box{index * sizeof(s32), 0, 0, (index + 1) * sizeof(s32), 1, 1};
-  D3D::context->UpdateSubresource(s_bbox_buffer.Get(), 0, &box, &value, 0, 0);
+  s_bbox_valid = false;
+
+  if (std::none_of(s_bbox_dirty.begin(), s_bbox_dirty.end(), [](bool dirty) { return dirty; }))
+    return;
+
+  for (u32 start = 0; start < NUM_BBOX_VALUES;)
+  {
+    if (!s_bbox_dirty[start])
+    {
+      start++;
+      continue;
+    }
+
+    u32 end = start + 1;
+    s_bbox_dirty[start] = false;
+    for (; end < NUM_BBOX_VALUES; end++)
+    {
+      if (!s_bbox_dirty[end])
+        break;
+
+      s_bbox_dirty[end] = false;
+    }
+
+    D3D11_BOX box{start * sizeof(s32), 0, 0, end * sizeof(s32), 1, 1};
+    D3D::context->UpdateSubresource(s_bbox_buffer.Get(), 0, &box, &s_bbox_values[start], 0, 0);
+  }
 }
 
-int BBox::Get(int index)
+void BBox::Readback()
 {
-  int data = 0;
   D3D::context->CopyResource(s_bbox_staging_buffer.Get(), s_bbox_buffer.Get());
+
   D3D11_MAPPED_SUBRESOURCE map;
   HRESULT hr = D3D::context->Map(s_bbox_staging_buffer.Get(), 0, D3D11_MAP_READ, 0, &map);
   if (SUCCEEDED(hr))
   {
-    data = ((s32*)map.pData)[index];
+    for (u32 i = 0; i < NUM_BBOX_VALUES; i++)
+    {
+      if (!s_bbox_dirty[i])
+      {
+        std::memcpy(&s_bbox_values[i], reinterpret_cast<const u8*>(map.pData) + sizeof(s32) * i,
+                    sizeof(s32));
+      }
+    }
+
+    D3D::context->Unmap(s_bbox_staging_buffer.Get(), 0);
   }
-  D3D::context->Unmap(s_bbox_staging_buffer.Get(), 0);
-  return data;
+
+  s_bbox_valid = true;
+}
+
+void BBox::Set(int index, int value)
```

Comment:
The `Set()` and `Get()` here don't do:

```
if (!g_ActiveConfig.backend_info.bSupportsBBox)
    return;
```

like OpenGL does.  Intentional?

Reply:
The check isn't needed for either backend. 

VideoCommon only requests Bounding Box reads when bounding box is enabled and supported by the backend.